saleor/graphql/product/mutations/product_variant/product_variant_update.py

```diff
@@ -142,6 +146,40 @@ def post_save_action(cls, info: ResolveInfo, instance, cleaned_input):
         ).values_list("channel_id", flat=True)
         cls.call_event(mark_active_catalogue_promotion_rules_as_dirty, channel_ids)
 
+    @classmethod
+    def _save(cls, info: ResolveInfo, instance, cleaned_input, base_fields_changed):
+        new_variant = instance.pk is None
+        cls.set_track_inventory(info, instance, cleaned_input)
+        variant_modified = False
+        with traced_atomic_transaction():
+            if base_fields_changed:
+                instance.save()
+                variant_modified = True
+            if not instance.product.default_variant:
+                instance.product.default_variant = instance
+                instance.product.save(update_fields=["default_variant", "updated_at"])
+            if stocks := cleaned_input.get("stocks"):
+                cls.create_variant_stocks(instance, stocks)
+                variant_modified = True
+            if attributes := cleaned_input.get("attributes"):
+                AttributeAssignmentMixin.save(instance, attributes)
+                variant_modified = True
+
+            if not instance.name:
+                generate_and_set_variant_name(instance, cleaned_input.get("sku"))
+                variant_modified = True
```

Comment:
Those changes will not be saved. (Save is in line 156.) 
IMO this code should be executed before calculating base_fields_changed.